src/compiler/core.ts

```diff
@@ -2035,11 +2035,22 @@ namespace ts {
      * Takes a string like "jquery-min.4.2.3" and returns "jquery"
      */
     export function removeMinAndVersionNumbers(fileName: string) {
+        // We used to use the regex /[.-]((min)|(\d+(\.\d+)*))$/ and would just .replace it twice.
+        // Unfortunately, that regex has O(n^2) performance because v8 doesn't match from the end of the string.
+        // Instead, we now split the input on the . and - characters, and drop the trailing segements
+        // which match the /^[.-](min|\d+)$/ pattern, which should be O(n). (This technically allows more `.min` repetitions than before, but that's fine)
+
         // Match a "." or "-" followed by a version number or 'min' at the end of the name
-        const trailingMinOrVersion = /[.-]((min)|(\d+(\.\d+)*))$/;
+        const trailingMinOrVersion = /^[.-](min|\d+)$/;
 
-        // The "min" or version may both be present, in either order, so try applying the above twice.
-        return fileName.replace(trailingMinOrVersion, "").replace(trailingMinOrVersion, "");
+        // Split on . and -, then remove all trailing segments that match
+        const parts = fileName.split(/(?=[.-])/g);
+        let end = parts.length - 1;
+        while (parts.length > 1 && end >= 0) {
+            if (!trailingMinOrVersion.test(parts[end])) break;
+            end--;
+        }
+        return parts.slice(0, end + 1).join("");

```

Comment:
Does this create a copy of the array and, if so, could that be avoided by setting `.length`?

Reply:
It does and that should work, yeah.